prov/rxm/src/rxm.h

```diff
@@ -421,10 +426,24 @@ struct rxm_conn {
 	struct dlist_entry posted_rx_list;
 	struct dlist_entry sar_rx_msg_list;
 	struct util_cmap_handle handle;
-	/* This is saved MSG EP fid, that hasn't been closed during
-	 * handling of CONN_RECV in CMAP_CONNREQ_SENT for passive side */
-	struct fid_ep *saved_msg_ep;
-	struct dlist_entry saved_posted_rx_list;
+
+	struct rxm_conn_close *close_data;
+};
+
+struct rxm_cmap_cmd_data {
+	/* common stuff */
+	struct rxm_conn *rxm_conn;
+
+	/* cmd-specific stuff */
+	union {
+		struct {
+			struct fi_info *msg_info;
+			struct rxm_cm_data cm_data;
+		} connreq_cmd;
+		struct {
+			struct rxm_conn_close *close_data;
```

Comment:
Can we avoid having to define a rxm_conn_close struct?

Reply:
Yes, we can, but we need to carry MSG_EP and posted_rx_list. What do you think about this approach?

Reply:
I have a PR in the works that eliminates the responsibility to maintaining posted_rx_list. IIUC, the list is needed to track posted receive buffers only to keep the util buffer pool happy in debug mode.

Reply:
Thanks, Arun! I've re-based this PR on top of your changes (master branch).

Reply:
rxm_close_data has been removed as no longer needed